### PR TITLE
Update substrate to polkadot-v0.9.24

### PR DESCRIPTION
**Pull request by gaudenzkessler**

### CODE REVIEW

**Review by haerdib — COMMENTED** [review]
LGTM! Though we shouldn't merge into master because of the branch name (master should stay on master in the api-client), but the branch looks good !

Are there any changes that need to be merged into master in this PR?

File: src/utils.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #[cfg(not(feature = "std"))]
 extern crate alloc;
 #[cfg(not(feature = "std"))]
-use alloc::{string::String, string::ToString, vec::Vec};
+use alloc::{string::String, vec::Vec};
```
[the remark this answers is not in the document]
Comment (echevrier) [reply src/utils.rs:21]: @gaudenzkessler Such changes should also come into master.

[the remark this answers is not in the document]
Comment (clangenb) [reply src/utils.rs:21]: Clippy does only check with std enabled, I believe

**Review by clangenb — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**